cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1459,6 +1460,11 @@ public List<ApplicationFilter> getAppFilters(Set<String> artifactNames,
    */
   public String decodeUserId(AuthenticationContext authenticationContext) {
     String decodedUserId = "emptyUserId";
+    // In case of nonrbac instances the principal name is not base64 encoded and its
+    // always cdap
+    if (authenticationContext.getPrincipal().getName().equals(NON_RBAC_USER_ID)) {
```

Comment:
Please use `Constants.Security.INTERNAL_AUTH_ENABLED` to identify whether RBAC is enabled or not.